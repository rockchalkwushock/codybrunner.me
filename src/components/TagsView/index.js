import React from 'react'

import { FlexContainer, Grid, Link, List, Tag } from '../commons'
import { sortedTags } from '../../lib/helpers'

// FIXME: Come up with a better display than this, it's garbage
// It also let's the footer jump up.
const Tags = ({ tags }) =>
  tags ? (
    <FlexContainer>
      <Grid>
        <h1>Tags</h1>
        <List tags>{sortedTags(tags, <Tag />)}</List>
      </Grid>
    </FlexContainer>
  ) : (
    <FlexContainer>
      <Grid>
        <p>Whoa warrior, woah! How did you do that? Maybe go home okay.</p>
        <Link href="/" text="Go back home" />
      </Grid>
    </FlexContainer>
  )

export default Tags
