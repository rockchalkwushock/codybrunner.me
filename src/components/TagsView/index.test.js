import React from 'react'
import { shallow } from 'enzyme'

import TagsView from './'

const tags = [['one', [1, 2, 3]], ['two', [1, 2, 3]], ['three', [1, 2, 3]]]

describe('Component: <TagsView />', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<TagsView tags={tags} />)
  })

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
  test('should have length of three', () => {
    expect(wrapper.find('List').children()).toHaveLength(3)
  })
})
