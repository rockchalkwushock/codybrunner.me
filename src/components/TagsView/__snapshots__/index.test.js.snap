// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <TagsView /> should render correctly 1`] = `
<FlexContainer>
  <Grid>
    <h1>
      Tags
    </h1>
    <List
      flow="row"
      gap={5}
      tags={true}
    >
      <Tag
        key="one"
        sortkey={3}
        to="/tags/one"
      >
        one  (3)
      </Tag>
      <Tag
        key="two"
        sortkey={3}
        to="/tags/two"
      >
        two  (3)
      </Tag>
      <Tag
        key="three"
        sortkey={3}
        to="/tags/three"
      >
        three  (3)
      </Tag>
    </List>
  </Grid>
</FlexContainer>
`;
